src/components/UserDashboard/charts.js

```diff
@@ -175,23 +164,41 @@ export const UserInterestsChart = withTheme(({ data, theme }) => {
   );
 });
 
-export const TopDiagnosesChart = compose(
+export const topDiagnoseChart = compose(
   withRouter,
   withTheme,
-)(({ data, theme, history }) => {
-  const mergedData = data.map(d => ({
-    ...d,
-    url: getFileRepoURL(TEXT_DIAGNOSES_FIELD, d.name),
-  }));
+)(({ data, theme, setSqons, virtualStudy, history }) => {
 
   const onClick = barData => {
     trackBarClick(diagnosesChartCategory, barData);
-    history.push(barData.data.url);
+    addSqon('diagnoses.diagnosis', barData.data.name);
+    history.push('/explore');
+  };
+
+  const addSqon = (field, value) => {
```

Comment:
Small thing: would extracting the addSqon function and making it top-level in the file a good idea? The addSqon function is also used at line 88, so we could avoid duplicated code this way. I don't know if this is the right move though as addSqon has side effects :)

Reply:
Would rather not since the components are now connected to the state.

Reply:
(passing the virtualstudy as a parameter would be required, though)

Reply:
Alright! LGTM then :)